### PR TITLE
 7.0.x: backport treating unknown requirements as unsatisfied with opt-out flag - v2

### PR DESCRIPTION
**Pull request by jasonish**
Replaces #12224 

Ticket: https://redmine.openinfosecfoundation.org/issues/7434

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/2162

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW2_autofp_stats_chk | | |
| .uptime | 145 | 140 | 96.55% | 

Pipeline 23679

File: doc/userguide/upgrade.rst [hunk 1]
```diff
@@ -34,6 +34,27 @@ also check all the new features that have been added but are not covered by
 this guide. Those features are either not enabled by default or require
 dedicated new configuration.
 
+Upgrading to 7.0.8
+------------------
+- Unknown requirements in the ``requires`` keyword will now be treated
+  as unsatisfied requirements, causing the rule to not be loaded. See
+  :ref:`keyword_requires`. To opt out of this change and to ignore
+  uknown requirements, effectively treating them as satified the
+  ``ignore-unknown-requirements`` configuration option can be used.
```
Comment (victorjulien) [inline doc/userguide/upgrade.rst:43]: are you thinking about making this a 7.0.x only option? If so, we should probably state that this is not behavior that will continue in 8

Comment (victorjulien) [reply doc/userguide/upgrade.rst:43]: if we want to keep this in 8 and beyond, I'd like it to live under `detect.`

File: src/detect-requires.c [hunk 2]
```diff
@@ -20,6 +20,10 @@
 #include "detect-engine.h"
 #include "rust.h"
 
+/* Set to true if unknown requirements should be ingored. In Suricata
+ * 8, unknown requirements are treated as unsatisfied requirements. */
```
Comment (victorjulien) [inline src/detect-requires.c:24]: ah I think this answers my question. So do we need a warning/doc reference about this being 7 only behavior?

**Review by victorjulien — COMMENTED** [review]
Code looks good. Have a question on the new option inline.

File: doc/userguide/upgrade.rst (hunk 1 above)
Comment (jasonish) [reply doc/userguide/upgrade.rst:43]: Ideally we wouldn't have it at all. So will add a note that it won't be around in 8.

Comment (jasonish) [reply doc/userguide/upgrade.rst:43]: > if we want to keep this in 8 and beyond, I'd like it to live under `detect.`

Also note the reasoning for the location: https://github.com/OISF/suricata/pull/12226#discussion_r1870844582

**Comment by jasonish** [comment]
Continued at https://github.com/OISF/suricata/pull/12239.

**Pull request closed without merging**